### PR TITLE
backport: bitcoin#15454, #16681, #16798, #16900, #18727, #19619, #20034, #20581, partial #17371, #18115 (berkeley db refactoring: part 3)

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
wow, that's a huge one! 👍 

pls see below + https://github.com/UdjinM6/dash/commits/pr4919 for some suggestions

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: doc/release-notes-15454.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Wallet
+------
+
+Dash Core will no longer create an unnamed `""` wallet by default when no wallet is specified on the command line or in the configuration files.
```
Comment (PastaPastaPasta) [inline doc/release-notes-15454.md:4]: thanks for dashifying this :)

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

File: src/wallet/test/wallet_tests.cpp [hunk 2]
```diff
@@ -31,6 +34,38 @@ extern UniValue getnewaddress(const JSONRPCRequest& request);
...
+    wallet->postInitProcess();
+    return wallet;
+}
+
+static void TestUnloadWallet(std::shared_ptr<CWallet>&& wallet)
+{
+    std::vector<bilingual_str> warnings;
+    SyncWithValidationInterfaceQueue();
+    wallet->m_chain_notifications_handler.reset();
+    RemoveWallet(wallet, nullopt, warnings);
```
Comment (knst) [inline src/wallet/test/wallet_tests.cpp:54]: anyone remembers, why it requires here `RemoveWallet` ?

https://github.com/bitcoin/bitcoin/pull/18727/files#diff-157f588a09ff1da05b8c74acfcfb4da21917b6a97ed5d2e702228d621d23e66dR40-R44

**Pull request merged**